### PR TITLE
Added flexible metrics generation from validation logits

### PR DESCRIPTION
**Pull request by JRosenkranz**
- added a way to capture different metrics from validation logits for flexibility of testing
- added a topk loss function to be used as a metrics calculator for capturing level 1 metrics
- added a generate metrics script which is used to compare fp32 cpu and bf16/fp16 cuda for use in generating baseline validation metrics

### CODE REVIEW

File: aiu_fms_testing_utils/testing/validation.py [hunk 1]
```diff
@@ -244,28 +244,59 @@ def validate_level_0(aiu_tokens_per_sentence, validation_tokens_per_sentence):
...
+        test_logits_prob = test_logits.to(dtype=torch.float32)
+
+        reference_values, reference_indices = torch.topk(reference_logits_prob, top_k, dim=1)
+        test_values = test_logits_prob[:, reference_indices.squeeze(0)]
+
+        return loss_f(reference_values, test_values)
+    return loss_func
+
+
+def capture_level_1_metrics(reference_logits_per_sentence, test_logits_per_sentence, metrics_calculator=None):
```
Comment (ani300) [inline aiu_fms_testing_utils/testing/validation.py:266]: is metrics_calulator the loss_function?
